packages/ui/src/button/Button.tsx

```diff
@@ -1,133 +1,93 @@
-  lightBlue: 'bg-blue-light',
-  lightViolet: 'bg-violet-light',
-  darkGreen: 'bg-green-dark',
-  darkRed: 'bg-red-dark',
-  darkBlue: 'bg-blue-dark',
-  darkViolet: 'bg-violet-dark',
-}
-
-const borderColors = {
-  green: 'border-green',
-  red: 'border-red',
-  blue: 'border-blue',
-  violet: 'border-violet',
-  lightGreen: 'border-green-light',
-  lightRed: 'border-red-light',
-  lightBlue: 'border-blue-light',
-  lightViolet: 'border-violet-light',
-  darkGreen: 'border-green-dark',
-  darkRed: 'border-red-dark',
-  darkBlue: 'border-blue-dark',
-  darkViolet: 'border-violet-dark',
-}
-
-const textColors = {
-  green: 'text-green',
-  red: 'text-red',
-  blue: 'text-blue',
-  violet: 'text-violet',
-  lightGreen: 'text-green-light',
-  lightRed: 'text-red-light',
-  lightBlue: 'text-blue-light',
-  lightViolet: 'text-violet-light',
-  darkGreen: 'text-green-dark',
-  darkRed: 'text-red-dark',
-  darkBlue: 'text-blue-dark',
-  darkViolet: 'text-violet-dark',
+const getColorClass = (attr: string, color: ButtonColorVariant) => {
+  const colorMap = {
+    green: 'green',
+    red: 'red',
+    blue: 'blue',
+    violet: 'violet',
+    lightGreen: 'green-light',
+    lightRed: 'red-light',
+    lightBlue: 'blue-light',
+    lightViolet: 'violet-light',
+    darkGreen: 'green-dark',
+    darkRed: 'red-dark',
+    darkBlue: 'blue-dark',
+    darkViolet: 'violet-dark',
+  }
+  return `${attr}-${colorMap[color]}`
 }
 
 const Button = ({
   text,
-  variant,
-  disabled,
-  loading,
-  size,
-  color,
+  variant = 'primary',
+  disabled = false,
+  warning = false,
+  loading = false,
+  size = 'M',
+  color = 'green',
   leftIcon,
   rightIcon,
-  style,
   onClick,
   className,
   children,
   ...rest
 }: Props) => {
   const sizeToClass = {
-    S: 'rounded-[16px] w-[100%] h-[32px] box-border',
-    M: 'rounded-[24px] w-[100%] h-[48px] box-border',
-    L: 'rounded-[26px] w-[100%] h-[52px] box-border',
+    S: 'rounded-[16px] h-[32px] note-3',
+    M: 'rounded-[24px] h-[48px] body-1',
+    L: 'rounded-[26px] h-[52px] title-1',
   }
+
+  const buttonColor = warning ? 'darkRed' : color
+
   const variantToClass = {
-    primary: `w-[100%] text-white box-border ${backgroundColors[color]}`,
-    outline: `w-[100%] ${textColors[color]} border ${borderColors[color]} bg-transparent`,
+    primary: `text-white ${getColorClass('bg', buttonColor)}`,
+    outline: `${getColorClass('text', buttonColor)} border ${getColorClass('border', buttonColor)} bg-transparent`,
   }
 
   const disabledStyleVariantToClass = {
     primary: `disabled:bg-gray-200`,
     outline: `disabled:text-gray-200 disabled:border-gray-200`,
   }
 
-  const textStyles = () => {
-    switch (size) {
-      case 'L':
-        return 'button-1'
-      case 'M':
-        return 'button-2'
-      case 'S':
-        return 'button-3'
-      default:
-        return 'button-1'
-    }
+  const IconComponent = ({ Icon }: { Icon: IconSource | undefined }) => {
+    if (!Icon) return null
```

Comment:
```suggestion
  const IconComponent = ({ Icon }: { Icon: IconSource) => {
```
Icon 이 props로 들어올때만 렌더링 하고 싶은거면 요렇게 변경하고 `IconComponent`를 호출하는 곳에서
```js
leftIcon && <IconComponent Icon={leftIcon} />
```
요렇게 하시믄 좀 더 명시적으로 언제 이 컴포넌트가 렌더링 되고 사용되는지 알 수 있을 것 같아여!